src/_guides/openapi/advanced-ref-usage.md

```diff
@@ -0,0 +1,335 @@
+        application/json:
+          schema:
+            $ref: ../components/schemas/status.yaml
+```
+
+Finally, `components/schemas/status.yaml` looks like this:
+
+```yaml
+type: object
+properties:
+  winner:
+    $ref: ./winner.yaml
+  board:
+    $ref: ./board.yaml
+```
+
+All the filepaths are relative to their current location, and can traverse up and down the filesystem, with the standard `../` to go up a directory.
+
+The chance of getting a Git conflict when two different developers add two different paths or expanding properties in a schema is now a fair bit smaller. If a conflict does occur, the diff will be a lot less confusing to work out.
+
+One downside of splitting up components into different documents like this is that it becomes harder to follow API changes, either directly or by looking at files in GitHub. Changing a schema in one document can effect how multiple different endpoints work, and that can caused a bit of confusion. API change management tools like [Bump.sh](https://bump.sh/api-change-management) or [Optic](https://www.useoptic.com/docs/diff-openapi) can help by spotting breaking changes and reporting them on PRs, so that you can easily see problems that could otherwise slip through.
+
+## AsyncAPI Reusable Components
+
+AsyncAPI is thankfully the same when it comes to `$ref` and `components`, so if your event-driven API is struggling as much as your HTTP API then it's time to split things up.
+
+```yaml
+  v0/rust/servers/{server_id}/players/{steam_id}/events/banned:
+    description: Channel for notifying a server banned a player
+    parameters:
+      server_id:
+        "$ref": "./components/parameters.json#/server_id"
+      steam_id:
+        "$ref": "./components/parameters.json#/steam_id"
+    subscribe:
+      operationId: ServerPlayerBanned
+      message:
+        "$ref": "./components/messages/ServerPlayerBanned.json"
+```
+
+This example is taken from the [Gaming API](https://github.com/GamingAPI/definitions/) example projects, and highlights a slightly different approach of using a single `parameters.json` document and referencing a parameter within that file, instead of using a `parameters/` subdirectory with a file for each parameter. You could do either with either OpenAPI or AsyncAPI, it's a matter of personal preference. 
+
+To learn more about the `components` keyword in AsyncAPI, head on over to [their documentation](https://www.asyncapi.com/docs/reference/specification/v2.6.0#componentsObject).
+
+## Using $ref with URLs
+
+Filepaths are not the only way to work with `$ref`, you can also use URLs.
+
+This is particularly helpful when you have a "data model" that is shared across multiple APIs or microservices. Perhaps you don't want each API to define a User, Company, or Payment separately, and get stuck with infinite different variant models. 
+
+Simply publish those shared components as JSON, YAML, or both, on a static site or S3 bucket somewhere and let people $ref them into their API.
```

Comment:
Should we also mention the possibility of sharing those files via a public git repository too?